src/core/sagas/autocomplete.ts

```diff
@@ -2,21 +2,66 @@ import * as effects from 'redux-saga/effects';
 import FluxCapacitor from '../../flux-capacitor';
 import Actions from '../actions';
 import Adapter from '../adapters/autocomplete';
+import Configuration from '../configuration';
 import Selectors from '../selectors';
 import Store from '../store';
+import { fetch } from '../utils';
 
 export namespace Tasks {
+  // tslint:disable-next-line max-line-length
   export function* fetchSuggestions(flux: FluxCapacitor, { payload: query }: Actions.FetchAutocompleteSuggestions) {
     try {
-      const field = yield effects.select(Selectors.autocompleteCategoryField);
-      const res = yield effects.call(
+      const state = yield effects.select();
+      const field = Selectors.autocompleteCategoryField(state);
+      const location = Selectors.location(state);
+      const suggestionsRequest = effects.call(
         [flux.clients.sayt, flux.clients.sayt.autocomplete],
         query,
         Selectors.autocompleteSuggestionsRequest(flux.config)
       );
-      const suggestions = Adapter.extractSuggestions(res, field);
+      const config = flux.config.autocomplete.recommendations;
+      const suggestionMode = Configuration.RECOMMENDATION_MODES[config.suggestionMode || 'popular'];
+      // tslint:disable-next-line max-line-length
+      const trendingUrl = `https://${flux.config.customerId}.groupbycloud.com/wisdom/v2/public/recommendations/searches/_get${suggestionMode}`;
+      const trendingBody: any = {
+        size: config.suggestionCount,
+        matchPartial: {
+          and: [{
+            search: { query }
+          }]
+        }
+      };
+      if (location && config.location) {
```

Comment:
determines whether or not the location should be requested in the first place by the autocomplete service, but I believe it's redundant here, yes